internal/03_publish_demo.Rmd

```diff
@@ -6,54 +6,73 @@ output:
   github_document:
     toc: true
 ---
+```{r setup, include=FALSE}
+library("googledrive")
+library("googlesheets")
+library("dplyr")
+library("readr")
+```
+
+
+## Motivation
+
+Push a table into a Sheet.
+
+Try to read it _as another user_ 
+Assume you even have the key.
+You will fail.
+
+Now, as the user who owns the Sheet, publish it.
 
-This is a little demo to show how we can check if a file is published & publish it if we so desire.
+Now, as the other user, try again to read it via googlesheets. You should succeed.
 
-```{r, message = FALSE, warning = FALSE}
-library('dplyr')
-library('googledrive')
+
+## Push a file into a Sheet
+```{r}
+drive_auth("drive-token.rds")
+drive_chickwts <- drive_upload("~/desktop/chickwts.csv", type = "spreadsheet")
 ```
 
+## Check publication status (should be FALSE)
 ```{r}
-write.table("This is a little demo", "demo.txt")
-drive_upload("demo.txt", name = "Happy Little Demo")
+drive_check_publish(drive_chickwts)
 ```
 
+## get URL
 ```{r}
-my_file <- drive_get_id("Happy Little Demo") %>%
-  drive_file
+url <- drive_share_link(drive_chickwts)
 ```
 
-## check published status
 
 ```{r}
-my_file %>%
-  drive_check_publish
+key <- drive_chickwts$id
 ```
-the `drive_check_publish` also outputs a new file with a tibble called `publish` with publication information. We can overwrite `my_file` to see this
-```{r}
-my_file <- my_file %>%
-  drive_check_publish
+
+## switch to different account
+```{r, eval = FALSE}
+gs_auth("sheets-token.rds")
 ```
 
+
+## this shouldn't work
 ```{r}
-my_file$publish
+try(gs_url(url, lookup = FALSE))
+geterrmessage()
 ```
 
-## publish it
+## publish it on Drive
 ```{r}
-my_file <- my_file %>%
-  drive_publish
+drive_chickwts <- drive_publish(drive_chickwts)
+drive_check_publish(drive_chickwts)
 ```
 
-now we have tibble with publication status
-
+## try again!
 ```{r}
-my_file$publish
+gs_url(url, lookup  = FALSE)
 ```
```

Comment:
Yes (I think) - so this file is an example of one that is accessible through the API, since it's published, but not shared with the world via link (in a browser). I can add the sharing permissions to the doc to show that as well. If I try to access this link in a browser (with my other google account) it claims I don't have permissions to.